src/components/menu-item/menu-item.tsx

```diff
@@ -0,0 +1,469 @@
+        }}
+        flipRtl={this.iconFlipRtl === "end" || this.iconFlipRtl === "both"}
+        icon={this.iconEnd}
+        scale="s"
+      />
+    );
+  }
+
+  renderBreadcrumbIcon(dir: Direction): VNode {
+    return (
+      <calcite-icon
+        class={{
+          [CSS.icon]: true,
+          [CSS.iconBreadcrumb]: true
+        }}
+        icon={dir === "rtl" ? "chevron-left" : "chevron-right"}
+        scale="s"
+      />
+    );
+  }
+
+  renderDropdownIcon(dir: Direction): VNode {
+    const dirChevron = dir === "rtl" ? "chevron-left" : "chevron-right";
+    return (
+      <calcite-icon
+        class={{
+          [CSS.icon]: true,
+          [CSS.iconDropdown]: true
+        }}
+        icon={
+          this.topLevelMenuLayout === "vertical" || this.isTopLevelItem
+            ? this.open
+              ? "chevron-up"
+              : "chevron-down"
+            : dirChevron
+        }
+        id="render-dropdown-icon"
+        scale="s"
+      />
+    );
+  }
+
+  renderDropdownAction(dir: Direction): VNode {
+    const dirChevron = dir === "rtl" ? "chevron-left" : "chevron-right";
+    return (
+      <calcite-action
+        class={CSS.dropdownAction}
+        icon={
+          this.topLevelMenuLayout === "vertical" || this.isTopLevelItem
+            ? this.open
+              ? "chevron-up"
+              : "chevron-down"
+            : dirChevron
+        }
+        onClick={this.clickHandler}
+        onKeyDown={this.keyDownHandler}
+        text="open-dropdown"
+        // eslint-disable-next-line react/jsx-sort-props
+        ref={(el) => (this.dropDownActionEl = el)}
+      />
+    );
+  }
+
+  renderSubMenuItems(dir: Direction): VNode {
+    return (
+      <calcite-menu
+        class={{
+          [CSS.dropdownMenuItems]: true,
+          [CSS.open]: this.open,
+          [CSS.nested]: !this.isTopLevelItem,
+          [CSS.isRtl]: dir === "rtl",
+          [CSS.dropdownVertical]: this.topLevelMenuLayout === "vertical"
+        }}
+        label="Submenu"
+        layout="vertical"
+        role="menu"
+      >
+        <slot name="sub-menu-item" onSlotchange={this.handleMenuItemSlotChange} />
+      </calcite-menu>
+    );
+  }
+
+  renderItemContent(dir: Direction): VNode {
+    return (
+      <Fragment>
+        {this.iconStart && this.renderIconElStart()}
+        <div class={CSS.textContainer}>
+          <span>{this.text}</span>
+        </div>
+        {!this.href && this.hasSubMenu ? this.renderDropdownIcon(dir) : null}
+        {this.breadcrumb ? this.renderBreadcrumbIcon(dir) : null}
+        {this.iconEnd && this.renderIconElEnd()}
+      </Fragment>
+    );
+  }
+
+  render() {
+    const dir = getElementDir(this.el);
+    return (
+      <Host onBlur={this.blurHandler} onFocus={this.focusHandler}>
+        <li
+          class={{
+            [CSS.container]: true,
+            [CSS.isParentVertical]: this.topLevelMenuLayout === "vertical"
+          }}
+          role="none"
```

Comment:
Why the role of none here?

Reply:
It is recommended as per w3spec that `li` in the submenu should have `role='none'` . More of a presentation element which will be ignored by assistive technologies.